ui/Model.elm

```diff
@@ -126,42 +154,113 @@ getReleaseList =
         |> Kinto.send ReleasesFetched
 
 
-extractChannels : List Release -> Dict.Dict Channel Bool
-extractChannels releaseList =
-    List.map (.details >> .channel >> (\c -> ( c, True ))) releaseList
+extractFilterSet : (Release -> String) -> List Release -> Dict.Dict String Bool
+extractFilterSet accessor releases =
+    List.map (accessor >> (\c -> ( c, True ))) releases
         |> Dict.fromList
 
 
-filterReleases : Model -> List Release
-filterReleases { filters, releases } =
+extractVersions : List Release -> Dict.Dict String Bool
+extractVersions releases =
+    -- TODO: sort desc
+    let
+        extractVersion ( version, x ) =
+            case (String.split "." version) of
+                v :: _ ->
+                    ( v, x )
+
+                _ ->
+                    ( "None", x )
+    in
+        extractFilterSet (.details >> .filename) releases
+            |> Dict.toList
+            |> List.map extractVersion
+            |> Dict.fromList
```

Comment:
This is unfortunate. There's no `Dict.mapKeys` in the Elm core library, but maybe we should try to add one. (There is one in a dict-extra package, but I don't like it for reasons which are outside the scope of this discussion.)